readthedocs/api/v2/views/model_views.py

```diff
@@ -336,6 +336,11 @@ def perform_create(self, serializer):
         build_api_key = self.request.build_api_key
         if not build_api_key.project.builds.filter(pk=build_pk).exists():
             raise PermissionDenied()
+
+        if BuildCommandResult.objects.filter(**serializer.validated_data).exists():
```

Comment:
I'd say this may be a slow query to run, maybe just limit the filtering to the build and start_time? Since we don't run commands concurrently, all of them should be guaranteed to be run sequentially (there may be an edge case with leap seconds, but that's a really really edge case :D).

Reply:
I don't think this query is gonna be slow as it's performed now:

```
In [12]: BuildCommandResult.objects.filter(**data).exists()
SELECT (1) AS "a"
  FROM "builds_buildcommandresult"
 WHERE ("builds_buildcommandresult"."build_id" = 7567631 AND "builds_buildcommandresult"."command" = 'git remote set-url origin https://github.com/failys/cairis.git' AND "builds_buildcommandresult"."description" = '' AND "builds_buildcommandresult"."end_time" = '2018-07-31T15:30:06.543902+00:00'::timestamptz AND "builds_buildcommandresult"."exit_code" = 0 AND "builds_buildcommandresult"."id" = 45588522 AND "builds_buildcommandresult"."output" = '' AND "builds_buildcommandresult"."start_time" = '2018-07-31T15:30:06.522766+00:00'::timestamptz)
 LIMIT 1

Execution time: 0.001359s [Database: default]
Out[12]: True
```

It takes 1.3ms in production currently.

Reply:
My worry is more about commands with very long outputs, we don't need to check the whole output, only the build id and the start date should be unambiguous.

Reply:
Good point about long outputs 👍🏼